src/game.ts

```diff
@@ -91,36 +69,42 @@ export const marbleWonByPlayer = (graph: Graph): number => {
   return -1;
 };
 
-export const renderScoreBoard = (players: Player[]): void => {
-  console.log("----------------");
-  console.log("Scoreboard");
-  for (const player of players) {
-    let playerScore = `Player ${player.playerNumber} :`;
-    for (const marble of player.marblesWon) {
-      playerScore += marbleValuetoANSIColorCode(marble) + " ";
+export const checkIfPlayerWon = (player: Player) => {
+  let otherPlayerMarbles = 0;
+  let neutralMarbles = 0;
+
+  for (const marble of player.marblesWon) {
+    if (marble !== player.marbleColor) {
+      if (marble === 3) {
+        neutralMarbles++;
+      } else {
+        otherPlayerMarbles++;
+      }
     }
-
-    console.log(playerScore);
   }
```

Comment:
You could use a reducer `player.marblesWon.reduce(...)`, but it's ok, at least you don't loop over the marbles twice ;)

Reply:
Also, I think you've missed something, I'm not sure if the blue player can push his own marble off the board, but if soo, you're counting them too in the 'otherPlayerMarbles'